src/test/java/com/google/devtools/build/lib/skylark/SkylarkRuleClassFunctionsTest.java

```diff
@@ -1624,6 +1625,34 @@ public void testRuleAddToolchain() throws Exception {
     assertThat(c.getRequiredToolchains()).containsExactly(makeLabel("//test:my_toolchain_type"));
   }
 
+  @Test
+  public void testRuleAddExecutionConstraints() throws Exception {
+    scratch.file("test/BUILD", "toolchain_type(name = 'my_toolchain_type')");
+    evalAndExport(
+        "def impl(ctx): return None",
+        "r1 = rule(impl, ",
+        "  toolchains=['//test:my_toolchain_type'],",
+        "  exec_compatible_with=['//constraint:cv1', '//constraint:cv2'],",
+        ")");
+    RuleClass c = ((SkylarkRuleFunction) lookup("r1")).getRuleClass();
+    assertThat(c.getExecutionPlatformConstraints())
+        .containsExactly(makeLabel("//constraint:cv1"), makeLabel("//constraint:cv2"));
+  }
+
+  @Test
+  public void testTargetsCanAddExecutionPlatformConstraints() throws Exception {
```

Comment:
Add a new test to illustrate what happens when a user defines the exec_compatible_with attribute themselves (using attrs = ...) -- whatever that behavior is supposed to be.

Reply:
I have added such a tes, but it doesn't seem to work, as this is only checked when RuleClass.Builder.build() is called, and for Skylark rules that happens when the rule is exported, and I can't figure out how to write a test that exports a rule. Any pointers?

Reply:
That's understandable. Check out SkylarkImportLookupFunction#execAndExport, and SkylarkRuleClassFunctionsTest#evalAndExport.

Reply:
Thanks, that worked!